### ISSUE TITLE
Corruption in the Log table

### ISSUE THREAD

**Issue by DaazKu** [vanilla/vanilla#5197]
We store a snapshot of the record being logged into the Data column.
If the serialized object has binary in it or if the object is too big we lose part of the record data.

See: 
- https://github.com/vanilla/vanilla/issues/3685
- https://github.com/vanilla/vanilla/issues/3715
- https://github.com/vanilla/vanilla/issues/2946

We need to sanitize the data properly before inserting in the table!
Otherwise we should try to update the column to BLOB instead of MEDIUMTEXT.

We will probably need to create a script to update our production databases.

**Comment by DaazKu** [comment]
#3715 speaks about binary data cutting string when inserting in the database. This was due to the IP columns being untranslated. There should be no binary data logged in the database. It's not possible to input binary data via the UI and the only binary data we have in the DB are the IPs.
#2946 is a problem about the data size which is caused by discussions being logged with their comments in one record.

After reviewing all the related task I came so the following conclusion:

- I will update `dbencode()` to now use `jsonEncodeChecked()`. (We will trace and log errors happening in dbencode).
- `dbdecode()` will be updated to be backward compatible.
- All the data column (Activity, Log, User) will remain as text columns since json_encode does not output binary data.
- Discussions will now log 1+(numbers of comments) in the log table. This will prevent the content from overflowing. (The [feature is already there](https://github.com/vanilla/vanilla/blob/dde933c808b6901ef7475e5cc48a33a08d6357f5/applications/vanilla/models/class.discussionmodel.php#L2732) but it's using an arbitrary number of comments)

**Comment by DaazKu** [comment]
TL;DR no db update, and everything will be JSON now.

**Comment by tburry** [comment]
There were long term plans to move dbencode() to json encoding so maybe you can see if there are issues there or ask @linc.

Another thing I'd like to see in terms of longer term support is for ipv6 addresses to show up as text *until* they go into the database. This means that models will need to encode/decode them into binary, but that encoding should be done at a low enough level that devs don't know that. This was a mistake and would have prevented us having to do all of this other stuff. To be clear though I'm okay with this being filed as an issue for later as it may have more of an impact on code in use.

**Comment by DaazKu** [comment]
An issue has already been filed it seems: https://github.com/vanilla/vanilla/issues/5323

**Comment by DaazKu** [comment]
Closing this in favor of tracking the issue on the PR https://github.com/vanilla/vanilla/pull/5797
